### PR TITLE
allow hash arguments

### PR DESCRIPTION
**Pull request by takkanm**
test all combinations of values.

You can be described as follows.
```
# use Traditional 

where(:a, :b, :c) do
  [[1, 5, 2], [1, 5, 4], [1, 7, 2], [1, 7, 4], [1, 9, 2], [1, 9, 4], [3, 5, 2], [3, 5, 4], [3, 7, 2], [3, 7, 4], [3, 9, 2], [3, 9, 4]]
end

# use Hash argument
 where(a: [1, 3], b: [5, 7, 9], c: [2, 4])
```

### CODE REVIEW

File: lib/rspec/parameterized.rb [hunk 1]
```diff
@@ -38,7 +38,17 @@ def initialize(arg_names, table_format, &block)
       #     end
       #
       def where(*args, &b)
-        set_parameters(args, false, &b)
+
+        if args.size == 1 && (params = args[0]).instance_of?(Hash)
```
Comment (joker1007) [inline lib/rspec/parameterized.rb]: `(params = args[0]).instance_of?(Hash)` is not reader friendly.
`params` assignment should be in `if` block

File: lib/rspec/parameterized.rb [hunk 2]
```diff
@@ -38,7 +38,17 @@ def initialize(arg_names, table_format, &block)
...
+          first, *rest = params.keys
+
+          set_parameters(params.keys, false) {
+            rest_values = rest.map {|k| params[k] }
+            params[first].product(*rest_values)
+          }
```
Comment (joker1007) [inline lib/rspec/parameterized.rb]:
```ruby
first, *rest = params.values
first.product(*rest)
```
I think this is better.
How do you feel?

Comment (takkanm) [inline lib/rspec/parameterized.rb]: :cool:

**Comment by takkanm** [comment]
I fix it.

**Comment by joker1007** [comment]
Thanks!!

**Pull request merged**